jme3-examples/src/main/java/jme3test/post/TestContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,101 @@
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package jme3test.post;
+
+import com.jme3.app.SimpleApplication;
+import com.jme3.light.AmbientLight;
+import com.jme3.material.Material;
+import com.jme3.math.ColorRGBA;
+import com.jme3.math.Vector2f;
+import com.jme3.math.Vector3f;
+import com.jme3.post.FilterPostProcessor;
+import com.jme3.post.filters.ContrastAdjustmentFilter;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.shape.Sphere;
+import com.jme3.texture.Texture;
+
+/**
+ * Tests {@link ContrastAdjustmentFilter} on different color channels with an
+ * adjustable exponents, scales and an input range during runtime.
+ * @author pavl_g.
+ */
+public class TestContrastAdjustmentFilter extends SimpleApplication {
+
+    private float counter = 0f;
+    private ContrastAdjustmentFilter contrastAdjustmentFilter;
+
+    public static void main(String[] args) {
+        new TestContrastAdjustmentFilter().start();
+    }
+
+    @Override
+    public void simpleInitApp() {
+        //setup a spatial and a texture
+        final Sphere globe = new Sphere(40, 40, 5f);
+        final Geometry earth = new Geometry("Globe", globe);
+        final Material material = new Material(assetManager, "Common/MatDefs/Misc/Unshaded.j3md");
+        final Texture texture = assetManager.loadTexture("Textures/Sky/Earth/Earth.jpg");
+        material.setTexture("ColorMap", texture);
+        earth.setMaterial(material);
+
+        rootNode.attachChild(earth);
+
+        //add light
+        final AmbientLight ambientLight = new AmbientLight(ColorRGBA.White);
+        rootNode.addLight(ambientLight);
+
+        //setup the filter
+        final FilterPostProcessor postProcessor = new FilterPostProcessor(assetManager);
+        contrastAdjustmentFilter = new ContrastAdjustmentFilter();
+        //adjusting some parameters
+        contrastAdjustmentFilter.setExponents(1.8f, 1.8f, 2.1f)
+                                .setInputRange(0, 0.367f)
+                                .setScales(0.25f, 0.25f, 1f);
+        postProcessor.addFilter(contrastAdjustmentFilter);
+        viewPort.addProcessor(postProcessor);
+
+    }
+
+    @Override
+    public void simpleUpdate(float tpf) {
+        counter += tpf;
+        if (counter >= 4f) {
+            counter = 0f;
+        }
+        if (contrastAdjustmentFilter != null) {
+            //adjust the transfer function during runtime
+            contrastAdjustmentFilter.setScales(counter * 0.5f, counter, counter * 0.5f)
+                                    .setExponents(counter, counter * 2f, counter)
+                                    .setInputRange(0, 1 / counter);
```

Comment:
What happens here when `counter` resets to 0? Looks to me like the input upper limit goes to `NaN`!

Reply:
Yep, I will limit that to 1.